### PR TITLE
Advanced Example Page

### PR DESCRIPTION
**Pull request by gkatsev**
This is a WIP. It still requires the PRs for playlist, playlist-ui, and contrib-hls to be merged in.
Also, I want to add more things to this example page following the examples from https://github.com/videojs/video.js/issues/2823

### CODE REVIEW

**Comment by gkatsev** [comment]
While it's good to go for review, it can't be released yet until we address https://github.com/videojs/videojs-contrib-hls/issues/688

File: _src-js/lib/bound-properties.js [hunk 1]
```diff
@@ -0,0 +1,136 @@
+import React from 'react';
+import ReactDOM from 'react-dom'
+import videojs from 'video.js';
+import shortid from 'shortid';
+
+const Form = React.createClass({
+  getInitialState() {
+    return {
```
Comment (mmcc) [inline _src-js/lib/bound-properties.js]: Not a big deal, just curious why you decided not to go with ES6 syntax for everything?

```javascript
class Form extends React.Component {
  constructor(props) {
    super(props);
    this.state = { ... };
  }

  ...
}
```

Comment (gkatsev) [inline _src-js/lib/bound-properties.js]: didn't realize you could do that. This is my first time using React and was just following their tutorial.
Though, totally makes sense that that should work.

Comment (mmcc) [inline _src-js/lib/bound-properties.js]: Yeah, it's a little annoying that they've deprecated some stuff (like mixins) and said that they suggest moving to ES2015 components like this....but then don't actually document very much in that way. There is one [quick blurb on them](https://facebook.github.io/react/docs/reusable-components.html#es6-classes), though.

**Comment by mmcc** [comment]
lgtm

File: _src-js/lib/bound-properties.js (hunk 1 above)
Comment (gkatsev) [inline _src-js/lib/bound-properties.js]: ah, cool. Didn't really look in other parts of the docs beyond the tutorial.
I might update it while waiting for the contrib-hls issue to resolve.

**Pull request merged**